evalml/data_checks/default_data_checks.py

```diff
@@ -43,28 +51,46 @@ class DefaultDataChecks(DataChecks):
         DateTimeNaNDataCheck,
     ]
 
-    def __init__(self, problem_type, objective, n_splits=3):
+    def __init__(self, problem_type, objective, n_splits=3, datetime_column="index"):
```

Comment:
Can we make the default None and if it is a time series problem, default to checking the index if datetime_column is not specified? I think that None makes it clearer that it's an optional param

Reply:
I can get behind this :)